app/components/BuyQuoteBox.tsx

```diff
@@ -0,0 +1,580 @@
+import { Button } from "@nextui-org/button";
+import { Input } from "@nextui-org/input";
+import {
+  ChangeEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { Card, Link, Tooltip } from "@nextui-org/react";
+import { Select, SelectItem } from "@nextui-org/select";
+import {
+  BuyConfigResponse,
+  BuyOptionsRequest,
+  BuyOptionsResponse,
+  BuyQuoteRequest,
+  BuyQuoteResponse,
+} from "../utils/types";
+import { generateBuyOptions, generateBuyQuote } from "../utils/queries";
+import ReactJson from "react-json-view";
+import { BuyConfigBox } from "./BuyConfigBox";
+import SecureTokenBox from "./SecureTokenBox";
+
+export default function BuyQuoteBox() {
+  /* refs to scroll to headers */
+  const buyOptionsHeaderRef = useRef<HTMLDivElement | null>(null);
+  const buyQuoteHeaderRef = useRef<HTMLDivElement | null>(null);
+
+  /* Buy Configuration Response State */
+  const [buyConfig, setBuyConfig] = useState<BuyConfigResponse>();
+
+  // buy quote loading state
+  // expand for the rest of the options or not -> disabled if prev section not complete (display tooltip)
+
+  /* Buy Options API Variables - Request & Response payloads, Loading state, List of country/subdiv options */
+  // request parameters and onChange wrapper function
+  const [buyOptionsParams, setBuyOptionsParams] = useState<BuyOptionsRequest>({
+    country: "",
+    subdivision: "",
+  });
+  const onChangeBuyOptionsParams = (e: ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = e.target;
+    setBuyOptionsParams((prevState) => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  // Buy Options API Response
+  const [buyOptionsResponse, setBuyOptionsResponse] =
+    useState<BuyOptionsResponse>();
+  const prevCountrySubdiv = useRef("");
+  const [isOptionsLoading, setIsOptionsLoading] = useState(false);
+
+  /* Change list of buy option country options on re-render when config generated */
+  const buy_options_countries = useMemo(() => {
+    return buyConfig?.countries || [];
+  }, [buyConfig]);
+  const buy_options_subdivisions = useMemo(() => {
+    return (
+      buy_options_countries
+        .find((country) => country.id === buyOptionsParams.country)
+        ?.subdivisions.map((s) => {
+          return { name: s };
+        }) || []
+    );
+  }, [buy_options_countries, buyOptionsParams.country]);
+
+  const emptyBuyQuoteParams = {
```

Comment:
Add a static type on this declaration

Reply:
i moved the declaration to before the component and explicility put its type as the buyQuoteRequest. Is that sufficient?